app/models/file_attachment/asset.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+# A FileAttachment::Asset is used to interface with Asset Manager to represent a
+# file uploaded as part of a FileAttachment::Revision. It is used to track the
+# state of the media on Asset Manager and its URL.
+#
+# This is a mutable model that is mutated when state changes on Asset Manager
+class FileAttachment::Asset < ApplicationRecord
+  belongs_to :file_revision,
+             class_name: "FileAttachment::FileRevision"
+
+  belongs_to :superseded_by,
```

Comment:
We'll only need this if we support replacing files, shall we assume we support replacing or add it in later?

Reply:
I've removed `superseded_by` from the model and the migration.